src/lxc/confile.c

```diff
@@ -487,6 +493,73 @@ static int set_config_net_veth_pair(const char *key, const char *value,
 	return network_ifname(netdev->priv.veth_attr.pair, value, sizeof(netdev->priv.veth_attr.pair));
 }
 
+static int set_config_net_veth_vlan_id(const char *key, const char *value,
+				  struct lxc_conf *lxc_conf, void *data)
+{
+	int ret;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return -1;
+
+	if (strcmp(value, "none") == 0) {
+		netdev->priv.veth_attr.vlan_id = BRIDGE_VLAN_NONE;
+	} else {
+		unsigned short vlan_id;
+		ret = get_u16(&vlan_id, value, 0);
+		if (ret < 0)
+			return -1;
+
+		if (vlan_id > BRIDGE_VLAN_ID_MAX)
+			return -1;
+
+		netdev->priv.veth_attr.vlan_id = vlan_id;
+	}
+
+	netdev->priv.veth_attr.vlan_id_set = true;
+	return 0;
+}
+
+static int set_config_net_veth_vlan_tagged_id(const char *key, const char *value,
+				       struct lxc_conf *lxc_conf, void *data)
+{
+	__do_free unsigned short *vlan_id = NULL;
+	__do_free struct lxc_list *list = NULL;
+	int ret;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_tagged_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return ret_set_errno(-1, EINVAL);
+
+	vlan_id = malloc(sizeof(*vlan_id));
+	if (!vlan_id)
+		return -1;
+
+	ret = get_u16(vlan_id, value, 0);
+	if (ret < 0)
+		return -1;
+
+	if (*vlan_id > BRIDGE_VLAN_ID_MAX)
+		return -1;
```

Comment:
ret_errno(EINVAL)

Reply:
In general, everywhere you can just use `ret_errno()`, use it. :)

Reply:
@brauner will do. What about the block above:

```
		ret = get_u16(&vlan_id, value, 0);
		if (ret < 0)
			return -1;
```

Is that suitable for use with `ret_errno(EINVAL)`? Or more generally what defines "everywhere you can just use `ret_errno()"? Do you have to check the function being called to see if it can set errno for you?

Reply:
Yes, if the function already returns a negative errno, you can just propagate it. If the function sets errno on error (usually library functions) you can return -errno.